source/Patches/CrewmateRoles/HaunterMod/CompleteTask.cs

```diff
@@ -0,0 +1,58 @@
+using System.Linq;
+using HarmonyLib;
+using Hazel;
+using TownOfUs.Roles;
+using Reactor;
+using UnityEngine;
+using TownOfUs.Extensions;
+
+namespace TownOfUs.CrewmateRoles.HaunterMod
+{
+    [HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.CompleteTask))]
+    public class CompleteTask
+    {
+        public static Sprite Sprite => TownOfUs.Arrow;
+        public static void Postfix(PlayerControl __instance)
+        {
+            if (!__instance.Is(RoleEnum.Haunter)) return;
+            var role = Role.GetRole<Haunter>(__instance);
+
+            var taskinfos = __instance.Data.Tasks.ToArray();
+
+            var tasksLeft = taskinfos.Count(x => !x.Complete);
+
+            if (tasksLeft == CustomGameOptions.HaunterTasksRemainingAlert && !role.Caught)
+            {
+                if (PlayerControl.LocalPlayer.Is(RoleEnum.Haunter))
+                {
+                    Coroutines.Start(Utils.FlashCoroutine(role.Color));
+                }
+                else if (PlayerControl.LocalPlayer.Data.IsImpostor() || (PlayerControl.LocalPlayer.Is(Faction.Neutral) && CustomGameOptions.HaunterRevealsNeutrals))

```

Comment:
How can this happen? Doesn't the code check to make sure they're the haunter first?